app/jobs/import_url_job.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+require Hyrax::Engine.root.join('app/jobs/import_url_job.rb')
+
+class ImportUrlJob
+  protected
+
+    def get_follow_redirects(uri, request_max = 5)
+      raise "Max number of redirects reached" if request_max <= 0
+
+      response = Net::HTTP.get_response(uri)
+      case response.code.to_i
+      when 200
+        uri
+      when 301..303
+        get_follow_redirects(URI(response['location']), request_max - 1)
+      else
+        response.code
```

Comment:
@hrotongn :  I don't know if I can test directly for a response.code in spec/jobs.  Everything I read says, that I can only use the have_http_status matcher in a controller, request, or feature spec.  I did add a success/failure test for the operation.  See below.

https://relishapp.com/rspec/rspec-rails/docs/matchers/have-http-status-matcher

Reply:
@scherztc Line #17 is still a problem.  The tests don't cover it and if the application hits that line it will crash the app because calling `get_follow_redirects` will return a string like "500" or "404" instead of a uri.

If there's no chance the application will ever hit that else condition, we should just remove it.  However, if a code other than 200 or 301..303 is possible, the method needs to gracefully deal with it instead of just returning `response.code`

You can get a test to hit that else condition by stubbing a 500 response and then calling .perform now:

In the before block
`WebMock.stub_request(:get, URI("http://example.org#{file_hash}")).to_return(
  body: File.open(File.expand_path(file_path, __FILE__)).read, status: 500, headers: { 'Content-Type' => 'image/png', 'location' => URI("http://example.org") }
)`

in the test:
`described_class.perform_now(file_set, operation)`

It crashes with exception:
```
Failure/Error:
       retriever.retrieve(spec) do |chunk|
         f.write(chunk)
       end

     NameError:
       undefined local variable or method `uri' for #<BrowseEverything::Retriever:0x00000007db5950 @chunk_size=16384>
       Did you mean?  url
                      URI
```

Reply:
@hortongn  I am looking into what Penn State did with Scholarsphere and trying to see if we can merge the two approaches.  https://github.com/psu-stewardship/scholarsphere/commit/22fe08c1957dda228bf4d6a5e239f6a48bee25f1#diff-27b3e499e0f5cc53eda5decc9879204d